### DISCUSSION TITLE
Qwen-Coder结束符是应该使用<endoftext>还是<im_end>?

### DISCUSSION THREAD

**Discussion by m-maoyanyu** [QwenLM/Qwen3#1104] [Q&A]
请问，Qwen-Coder的config.json与generation.json中eos_token_id均为151643："<|endoftext|>"，但加载Qwen-Coder的tokenizer后，eos_token_id为151645："<|endoftext|>" ："<|im_end|>"。直接使用Qwen-Coder推理时，也是输出"<|im_end|>"结尾。请问要以哪个作为结束符更好。
![image](https://github.com/user-attachments/assets/0cde8bbb-70f5-4375-ad26-0db11c2482ed)
![image](https://github.com/user-attachments/assets/ce53b799-a7f0-47f8-896f-cce7515a7860)

**Comment by quanshr** [comment]
base 用 <|endoftext|>；instruct 用 <|im_end|>
